core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/StructuredSyncEventFactory.java

```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.cloudbreak.structuredevent;
+
+import static com.sequenceiq.cloudbreak.structuredevent.event.StructuredEventType.SYNC;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Component;
+
+import com.cloudera.cdp.shaded.org.apache.commons.lang3.exception.ExceptionUtils;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.StackType;
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredSyncEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.SyncDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.legacy.OperationDetails;
+import com.sequenceiq.flow.ha.NodeConfig;
+
+@Component
+@Transactional
+public class StructuredSyncEventFactory {
+
+    @Inject
+    private Clock clock;
+
+    @Inject
+    private ConversionService conversionService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private NodeConfig nodeConfig;
+
+    @Value("${info.app.version:}")
+    private String serviceVersion;
+
+    public StructuredSyncEvent createStructuredSyncEvent(Long resourceId) {
+        return createStructuredSyncEvent(resourceId, null);
+    }
+
+    public StructuredSyncEvent createStructuredSyncEvent(Long resourceId, Exception exception) {
```

Comment:
are we use the method with exception parameter? if not then we can remove it and then we can remove the if statement below